build_local.sh

```diff
@@ -8,6 +8,13 @@ set -o errexit -o pipefail
 # Fail quickly if docker isn't working / up
 docker ps
 
+# Also fail quickly if Docker doesn't have enough memory to build Mesos
+DOCKER_MEMORY=$(docker info | grep Memory | awk '{print $3}')
+if (( $(echo "$DOCKER_MEMORY < 5" | bc -l) )); then
```

Comment:
@cmaloney do you want to target this to only OS X or keep it as a general check?